src/app.js

```diff
@@ -136,27 +301,36 @@ fastify.get('/viewer/rs/studies/:studyInstanceUid/series/:seriesInstanceUid/meta
   const { query } = req;
   query.StudyInstanceUID = req.params.studyInstanceUid;
   query.SeriesInstanceUID = req.params.seriesInstanceUid;
-  emitToWsClient(reply, 'IMAGE', query);
+  return emitToWsClient(reply, 'IMAGE', query);
 });
 
 //------------------------------------------------------------------
 
-fastify.get('/viewer/wadouri', async (req, reply) => {
+fastify.get('/viewer/wadouri', (req, reply) => new Promise((resolve) => {
   const uuid = uuidv4();
 
-  connectedClient.once(uuid, (data) => {
-    reply.header('Content-Type', 'application/dicom');
-    reply.send(data);
-  });
-  const { query } = req;
-  connectedClient.emit('wadouri-request', { query, uuid });
-});
+  if (connectedClient) {
+    connectedClient.once(uuid, (data) => {
+      reply.header('Content-Type', data.contentType);
+      reply.send(data.buffer);
+      resolve();
```

Comment:
can I just ask why this was needed?

Reply:
Because the response is in a total different format to wado-rs and qido-rs, so it's handled separately. The response is in the required format for dicomweb though